packages/api/lambdas/create-reconciliation-report.js

```diff
@@ -715,77 +734,93 @@ async function createReconciliationReport(recReportParams) {
+      );
 
-    const bucketReports = await Promise.all(promisedBucketReports);
-    log.info('bucketReports (S3 vs database) completed');
+      const bucketReports = await Promise.all(promisedBucketReports);
+      log.info('bucketReports (S3 vs database) completed');
 
-    bucketReports.forEach((bucketReport) => {
-      report.filesInCumulus.okCount += bucketReport.okCount;
-      report.filesInCumulus.onlyInS3 = report.filesInCumulus.onlyInS3.concat(bucketReport.onlyInS3);
-      report.filesInCumulus.onlyInDb = report.filesInCumulus.onlyInDb.concat(
-        bucketReport.onlyInDb
-      );
+      bucketReports.forEach((bucketReport) => {
+        report.filesInCumulus.okCount += bucketReport.okCount;
+        report.filesInCumulus.onlyInS3 = report.filesInCumulus.onlyInS3.concat(bucketReport.onlyInS3); // eslint-disable-line max-len
+        report.filesInCumulus.onlyInDb = report.filesInCumulus.onlyInDb.concat(
+          bucketReport.onlyInDb
+        );
 
-      if (linkingFilesToGranules(recReportParams.reportType)) {
-        Object.keys(bucketReport.okCountByGranule).forEach((granuleId) => {
-          const currentGranuleCount = report.filesInCumulus.okCountByGranule[granuleId];
-          const bucketGranuleCount = bucketReport.okCountByGranule[granuleId];
+        if (linkingFilesToGranules(recReportParams.reportType)) {
+          Object.keys(bucketReport.okCountByGranule).forEach((granuleId) => {
+            const currentGranuleCount = report.filesInCumulus.okCountByGranule[granuleId];
+            const bucketGranuleCount = bucketReport.okCountByGranule[granuleId];
 
-          report.filesInCumulus.okCountByGranule[granuleId] = (currentGranuleCount || 0)
-            + bucketGranuleCount;
-        });
-      } else {
-        delete report.filesInCumulus.okCountByGranule;
-      }
-    });
-  }
+            report.filesInCumulus.okCountByGranule[granuleId] = (currentGranuleCount || 0)
+              + bucketGranuleCount;
+          });
+        } else {
+          delete report.filesInCumulus.okCountByGranule;
+        }
+      });
+    }
 
-  // compare the CUMULUS holdings with the holdings in CMR
-  // -----------------------------------------------------
-  if (location !== 'S3') {
-    const cumulusCmrReport = await reconciliationReportForCumulusCMR({
-      bucketsConfig, distributionBucketMap, recReportParams,
-    });
-    report = Object.assign(report, cumulusCmrReport);
+    // compare the CUMULUS holdings with the holdings in CMR
+    // -----------------------------------------------------
+    if (location !== 'S3') {
+      const cumulusCmrReport = await reconciliationReportForCumulusCMR({
+        bucketsConfig, distributionBucketMap, recReportParams,
+      });
+      report = Object.assign(report, cumulusCmrReport);
+    }
+  } catch (error) {
+    log.error(`Error caught in createReconciliationReport for reportKey ${reportKey}`);
```

Comment:
> Can we move this logic to here for all reports?

If I moved the logic there, then there would be no report to reference and I would need to return the report object or retrieve it from S3. 

I updated to write the failed report to s3 because in the beginning of `createReconciliationReport` we write the report to S3 https://github.com/nasa/cumulus/pull/3332/files/676b39531abf20b2fc27d988e1d582cbcded46fe#diff-61b4ede5087a45c0d1191b106bc92b405c40e436ec41bf30bd9b86ef5bb2d634R738 but if we are okay with the report having no status if report generation fails, then I can remove this: 
https://github.com/nasa/cumulus/pull/3332/files/676b39531abf20b2fc27d988e1d582cbcded46fe#diff-61b4ede5087a45c0d1191b106bc92b405c40e436ec41bf30bd9b86ef5bb2d634R783-R790

What do you think @jennyhliu ?

Reply:
I think we need to write the failed report to S3 since that was the original behavior. I think it's also important to keep in mind that before, we didn't throw any errors so we would always write to S3 even when the report generation failed. Now, since we throw an error, we need to make sure to update the report in S3, otherwise the report status will be 'RUNNING'. @jennyhliu

Reply:
ok make sense, thanks for explanation